packages/client/src/components/ModuleTree/Module/Module.tsx

```diff
@@ -1,25 +1,104 @@
+
+import { ModuleIcon } from './Icons/Icon';
 
 export interface ModuleProps {
-  module: EModule
+  module: EModule,
+  myModule?: UserModule,
+  myConcepts?: UserConcept[]
 }
 
 export const Module: React.FC<ModuleProps> = ({
-  module
-}) => <div className={styles.module}>
-  <div className="relative">
-    <Hex
-      className={classnames('h-12 stroke-2', {
-        // TODO: Add completed to module
-        completed: false
-      })}
-    />
-    <Icon icon="fire" size={4} />
-  </div>
-  <span className="text-xs">{module.name}</span>
-</div>;
+  module,
+  myModule,
+  myConcepts
+}) => {
+  const isCompleted = myModule?.completedAt;
+  const isOpen = myModule && !myModule.completedAt;
+  const isLocked = !myModule;
+
+  return <div style={{
+    textAlign: 'center',
+    marginBottom: '1.5rem',
+    color: theme.colors.primary['500'] }}
+  >
+
+    <div style={{
+      position: 'relative',
+      display: 'flex',
+      height: '3rem',
+      width: '3rem',
+      marginRight: 'auto',
+      marginLeft: 'auto' }}
+    >
+
+      <ModuleIcon
+        icon={module.type === 'lesson' ? 'hexagon' : 'octagon'}
+        position="absolute"
+        size={3}
+        // eslint-disable-next-line no-nested-ternary
+        fill={isCompleted ? theme.colors.secondary['500'] : isOpen ? theme.colors.primary['400'] : theme.colors.white}
+        stroke={isLocked ? theme.colors.grey['500'] : undefined}
+        left="50%"
+        top="50%"
+        transform="translate(-50%, -50%)"
+      />
+
+      { module.type !== 'assignment'
+      && <ModuleIcon
+        icon="circle"
+        position="absolute"
+        size="small"
+        // eslint-disable-next-line no-nested-ternary
+        fill={isCompleted ? theme.colors.secondary['500'] : isOpen ? theme.colors.primary['400'] : theme.colors.grey['500']}
+        stroke={isLocked ? undefined : theme.colors.white}
+        left="85%"
+        top="20%"
+        transform="translate(-50%, -50%)"
+      /> }
+
+      {/* completed status with check mark in the circle. */}
+      { module.type !== 'assignment'
+      && isCompleted
+      && <ModuleIcon
+        icon="check"
+        position="absolute"
+        size={0.75}
+        color={theme.colors.white}
+        fill={theme.colors.white}
+        left="85%"
+        top="20%"
+        transform="translate(-50%, -50%)"
+      />}
+
+      {/* open status with number of completed concepts in the circle. */}
+      { module.type !== 'assignment'
+      && isOpen
+      && <span style={{
+        left: '80%',
+        top: '7%',
+        position: 'absolute',
+        fontSize: '0.5rem',
+        color: theme.colors.white }}
+      >{myConcepts?.length}</span> }
+
+      <ModuleIcon
+        icon="fire"
+        position="absolute"
+        size="small"
+        fill={isLocked ? theme.colors.grey['500'] : theme.colors.white}
+        left="50%"
+        top="50%"
+        transform="translate(-50%, -50%)"
+      />
+    </div>
+    <span style={{
```

Comment:
I think we shouldn't use inline styles and use either styled components (first option) or class names (second)

Reply:
changed to styled-component. 
commit: fix: change inline style to styled-components 31d9a1e